installation process failed

I tried to install this driver (because I got the Blackwidow Chroma) and it seems that some SDL libraries are missing. When I execute the install_driver.sh file, I get this:


In file included from razer_chroma.c:1:0:
razer_chroma.h:17:22: fatal error: SDL2/SDL.h: Datei oder Verzeichnis nicht gefunden
 #include "SDL2/SDL.h"
                      ^
compilation terminated.
make[1]: *** [razer_chroma.o] Fehler 1
make: *** [librazer_chroma] Fehler 2
In file included from razer_chroma.c:1:0:
razer_chroma.h:17:22: fatal error: SDL2/SDL.h: Datei oder Verzeichnis nicht gefunden
 #include "SDL2/SDL.h"
                      ^
compilation terminated.
make[1]: *** [razer_chroma.o] Fehler 1
make: *** [librazer_chroma] Fehler 2



"Datei oder Verzeichnis nicht gefunden" is german for "File or Folder is not found"
"Fehler" is german for "Error"
(for all those guys that don't understand german ^^)

You should also add the dependencies of this driver to the description... :smiley:

I'm on a fresh arch-linux install. After installing relevant packages (sdl2, sdl2_gfx, sdl2_image, sdl2_mixer), I attemted to build the driver with the Install_driver.sh:

```=====================================================
::::::::::::::::\033[32m COMPILING librazer_chroma.so\033[0m ::::::::::::::
=====================================================
=====================================================
::::::::::::::::\033[32m COMPILING librazer_chroma.so [DEBUG]\033[0m ::::::::::::::
=====================================================
=====================================================
::::::::::::::::\033[32m COMPILING razer_bcd [SHARED]\033[0m ::::::::::::::
=====================================================
=====================================================
::::::::::::::::\033[32m COMPILING razer_bcd daemon\033[0m ::::::::::::::
=====================================================
=====================================================
::::::::::::::::\033[32m COMPILING razer_bcd daemon [DEBUG]\033[0m ::::::::::::::
=====================================================
In file included from razer_daemon.c:1:0:
razer_daemon.h:21:41: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_open)(struct razer_fx_render_node *render);
                                         ^
razer_daemon.h:21:41: warning: its scope is only this definition or declaration, which is probably not what you want
razer_daemon.h:22:42: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_close)(struct razer_fx_render_node *render);
                                          ^
razer_daemon.h:23:43: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_update)(struct razer_fx_render_node *render);
                                           ^
razer_daemon.h:24:46: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_key_event)(struct razer_fx_render_node *render);
                                              ^
razer_daemon.h:25:47: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_dbus_event)(struct razer_fx_render_node *render);
                                               ^
razer_daemon.h:27:58: warning: ‘struct razer_daemon’ declared inside parameter list
 typedef struct razer_effect *(*razer_effect_init)(struct razer_daemon *daemon);
                                                          ^
razer_daemon.c: In function ‘dbus_handle_messages’:
razer_daemon.c:85:11: warning: passing argument 1 of ‘printf’ from incompatible pointer type [-Wincompatible-pointer-types]
    printf(stderr, "Out Of Memory!\n");
           ^
In file included from razer_daemon.h:4:0,
                 from razer_daemon.c:1:
/usr/include/stdio.h:362:12: note: expected ‘const char * restrict’ but argument is of type ‘struct _IO_FILE *’
 extern int printf (const char *__restrict __format, ...);
            ^
razer_daemon.c:117:11: warning: passing argument 1 of ‘printf’ from incompatible pointer type [-Wincompatible-pointer-types]
    printf(stderr, "Out Of Memory!\n");
           ^
In file included from razer_daemon.h:4:0,
                 from razer_daemon.c:1:
/usr/include/stdio.h:362:12: note: expected ‘const char * restrict’ but argument is of type ‘struct _IO_FILE *’
 extern int printf (const char *__restrict __format, ...);
            ^
razer_daemon.c:167:11: warning: passing argument 1 of ‘printf’ from incompatible pointer type [-Wincompatible-pointer-types]
    printf(stderr, "Out Of Memory!\n");
           ^
In file included from razer_daemon.h:4:0,
                 from razer_daemon.c:1:
/usr/include/stdio.h:362:12: note: expected ‘const char * restrict’ but argument is of type ‘struct _IO_FILE *’
 extern int printf (const char *__restrict __format, ...);
            ^
razer_daemon.c: In function ‘daemon_run’:
razer_daemon.c:240:3: warning: implicit declaration of function ‘effect3’ [-Wimplicit-function-declaration]
   effect3(daemon);
   ^
razer_daemon.c:244:3: warning: implicit declaration of function ‘razer_frame_limiter’ [-Wimplicit-function-declaration]
   razer_frame_limiter(daemon->chroma,daemon->fps);
   ^
razer_daemon.c:245:17: warning: unused variable ‘end_ticks’ [-Wunused-variable]
   unsigned long end_ticks = razer_get_ticks();
                 ^
razer_daemon.c:239:17: warning: unused variable ‘ticks’ [-Wunused-variable]
   unsigned long ticks = razer_get_ticks();
                 ^
razer_daemon.c: At top level:
razer_daemon.c:317:6: warning: conflicting types for ‘effect3’
 void effect3(struct razer_daemon *daemon)
      ^
razer_daemon.c:240:3: note: previous implicit declaration of ‘effect3’ was here
   effect3(daemon);
   ^
razer_daemon.c: In function ‘effect3’:
razer_daemon.c:320:14: warning: unused variable ‘fb’ [-Wunused-variable]
  float fr,fg,fb;
              ^
razer_daemon.c:320:11: warning: unused variable ‘fg’ [-Wunused-variable]
  float fr,fg,fb;
           ^
razer_daemon.c:320:8: warning: unused variable ‘fr’ [-Wunused-variable]
  float fr,fg,fb;
        ^
razer_daemon.c: In function ‘main’:
razer_daemon.c:995:14: warning: unused parameter ‘argc’ [-Wunused-parameter]
 int main(int argc,char *argv[])
              ^
razer_daemon.c:995:25: warning: unused parameter ‘argv’ [-Wunused-parameter]
 int main(int argc,char *argv[])
                         ^
razer_daemon.c: In function ‘dbus_handle_messages’:
razer_daemon.c:193:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
razer_daemon.c: In function ‘daemon_run’:
razer_daemon.c:248:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
In file included from razer_daemon.c:1:0:
razer_daemon.h:21:41: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_open)(struct razer_fx_render_node *render);
                                         ^
razer_daemon.h:21:41: warning: its scope is only this definition or declaration, which is probably not what you want
razer_daemon.h:22:42: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_close)(struct razer_fx_render_node *render);
                                          ^
razer_daemon.h:23:43: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_update)(struct razer_fx_render_node *render);
                                           ^
razer_daemon.h:24:46: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_key_event)(struct razer_fx_render_node *render);
                                              ^
razer_daemon.h:25:47: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_dbus_event)(struct razer_fx_render_node *render);
                                               ^
razer_daemon.h:27:58: warning: ‘struct razer_daemon’ declared inside parameter list
 typedef struct razer_effect *(*razer_effect_init)(struct razer_daemon *daemon);
                                                          ^
razer_daemon.c: In function ‘dbus_handle_messages’:
razer_daemon.c:85:11: warning: passing argument 1 of ‘printf’ from incompatible pointer type [-Wincompatible-pointer-types]
    printf(stderr, "Out Of Memory!\n");
           ^
In file included from razer_daemon.h:4:0,
                 from razer_daemon.c:1:
/usr/include/stdio.h:362:12: note: expected ‘const char * restrict’ but argument is of type ‘struct _IO_FILE *’
 extern int printf (const char *__restrict __format, ...);
            ^
razer_daemon.c:117:11: warning: passing argument 1 of ‘printf’ from incompatible pointer type [-Wincompatible-pointer-types]
    printf(stderr, "Out Of Memory!\n");
           ^
In file included from razer_daemon.h:4:0,
                 from razer_daemon.c:1:
/usr/include/stdio.h:362:12: note: expected ‘const char * restrict’ but argument is of type ‘struct _IO_FILE *’
 extern int printf (const char *__restrict __format, ...);
            ^
razer_daemon.c:167:11: warning: passing argument 1 of ‘printf’ from incompatible pointer type [-Wincompatible-pointer-types]
    printf(stderr, "Out Of Memory!\n");
           ^
In file included from razer_daemon.h:4:0,
                 from razer_daemon.c:1:
/usr/include/stdio.h:362:12: note: expected ‘const char * restrict’ but argument is of type ‘struct _IO_FILE *’
 extern int printf (const char *__restrict __format, ...);
            ^
razer_daemon.c: In function ‘daemon_run’:
razer_daemon.c:240:3: warning: implicit declaration of function ‘effect3’ [-Wimplicit-function-declaration]
   effect3(daemon);
   ^
razer_daemon.c:244:3: warning: implicit declaration of function ‘razer_frame_limiter’ [-Wimplicit-function-declaration]
   razer_frame_limiter(daemon->chroma,daemon->fps);
   ^
razer_daemon.c:245:17: warning: unused variable ‘end_ticks’ [-Wunused-variable]
   unsigned long end_ticks = razer_get_ticks();
                 ^
razer_daemon.c:239:17: warning: unused variable ‘ticks’ [-Wunused-variable]
   unsigned long ticks = razer_get_ticks();
                 ^
razer_daemon.c: At top level:
razer_daemon.c:317:6: warning: conflicting types for ‘effect3’
 void effect3(struct razer_daemon *daemon)
      ^
razer_daemon.c:240:3: note: previous implicit declaration of ‘effect3’ was here
   effect3(daemon);
   ^
razer_daemon.c: In function ‘effect3’:
razer_daemon.c:320:14: warning: unused variable ‘fb’ [-Wunused-variable]
  float fr,fg,fb;
              ^
razer_daemon.c:320:11: warning: unused variable ‘fg’ [-Wunused-variable]
  float fr,fg,fb;
           ^
razer_daemon.c:320:8: warning: unused variable ‘fr’ [-Wunused-variable]
  float fr,fg,fb;
        ^
razer_daemon.c: In function ‘main’:
razer_daemon.c:995:14: warning: unused parameter ‘argc’ [-Wunused-parameter]
 int main(int argc,char *argv[])
              ^
razer_daemon.c:995:25: warning: unused parameter ‘argv’ [-Wunused-parameter]
 int main(int argc,char *argv[])
                         ^
razer_daemon.c: In function ‘dbus_handle_messages’:
razer_daemon.c:193:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
razer_daemon.c: In function ‘daemon_run’:
razer_daemon.c:248:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
::\033[32m COMPILING pez2001 fx collection [DEBUG]\033[0m
========================================================
gcc: warning: ../../lib/librazer_chroma.da: linker input file unused because linking not done
::\033[32m COMPILING pez2001 fx collection\033[0m
=====================================================
gcc: warning: ../../lib/librazer_chroma.da: linker input file unused because linking not done
=====================================================
::::::::::::::::\033[32m COMPILING examples [DEBUG]\033[0m ::::::::::::::
=====================================================
=====================================================
::::::::::::::::\033[32m COMPILING examples\033[0m ::::::::::::::
=====================================================
=====================================================
::::::::::::::::\033[32m COMPILING razer chroma kernel module\033[0m ::::::::::::::
=====================================================
In file included from include/linux/thread_info.h:11:0,
                 from ./arch/x86/include/asm/preempt.h:6,
                 from include/linux/preempt.h:18,
                 from include/linux/spinlock.h:50,
                 from include/linux/mmzone.h:7,
                 from include/linux/gfp.h:5,
                 from include/linux/slab.h:14,
                 from /tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:23:
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:654:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(mode_wave, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_wave, razer_attr_write_mode_wave);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:655:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(mode_spectrum, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_spectrum, razer_attr_write_mode_spectrum);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:656:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(mode_none, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_none, razer_attr_write_mode_none);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:657:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(mode_reactive, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_reactive, razer_attr_write_mode_reactive);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:658:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(mode_breath, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_breath, razer_attr_write_mode_breath);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:659:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(mode_custom, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_custom, razer_attr_write_mode_custom);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:660:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(mode_static, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_static, razer_attr_write_mode_static);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:661:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(temp_clear_row, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_temp_clear_row, razer_attr_write_temp_clear_row);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:662:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(set_key_row, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_set_key_row, razer_attr_write_set_key_row);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:663:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(reset, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_reset, razer_attr_write_reset);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:664:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(macro_keys, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_macro_keys, razer_attr_write_macro_keys);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:665:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(set_brightness, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_set_brightness, razer_attr_write_set_brightness);
        ^
scripts/Makefile.build:264: recipe for target '/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.o' failed
make[2]: *** [/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.o] Error 1
Makefile:1390: recipe for target '_module_/tmp/razer_blackwidow_chroma_driver/driver' failed
make[1]: *** [_module_/tmp/razer_blackwidow_chroma_driver/driver] Error 2
Makefile:6: recipe for target 'all' failed
make: *** [all] Error 2
=====================================================
::::::::::::::::\033[32m COMPILING librazer_chroma.so\033[0m ::::::::::::::
=====================================================
=====================================================
::::::::::::::::\033[32m COMPILING librazer_chroma.so [DEBUG]\033[0m ::::::::::::::
=====================================================
=====================================================
::::::::::::::::\033[32m COMPILING razer_bcd [SHARED]\033[0m ::::::::::::::
=====================================================
=====================================================
::::::::::::::::\033[32m COMPILING razer_bcd daemon\033[0m ::::::::::::::
=====================================================
=====================================================
::::::::::::::::\033[32m COMPILING razer_bcd daemon [DEBUG]\033[0m ::::::::::::::
=====================================================
In file included from razer_daemon.c:1:0:
razer_daemon.h:21:41: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_open)(struct razer_fx_render_node *render);
                                         ^
razer_daemon.h:21:41: warning: its scope is only this definition or declaration, which is probably not what you want
razer_daemon.h:22:42: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_close)(struct razer_fx_render_node *render);
                                          ^
razer_daemon.h:23:43: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_update)(struct razer_fx_render_node *render);
                                           ^
razer_daemon.h:24:46: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_key_event)(struct razer_fx_render_node *render);
                                              ^
razer_daemon.h:25:47: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_dbus_event)(struct razer_fx_render_node *render);
                                               ^
razer_daemon.h:27:58: warning: ‘struct razer_daemon’ declared inside parameter list
 typedef struct razer_effect *(*razer_effect_init)(struct razer_daemon *daemon);
                                                          ^
razer_daemon.c: In function ‘dbus_handle_messages’:
razer_daemon.c:85:11: warning: passing argument 1 of ‘printf’ from incompatible pointer type [-Wincompatible-pointer-types]
    printf(stderr, "Out Of Memory!\n");
           ^
In file included from razer_daemon.h:4:0,
                 from razer_daemon.c:1:
/usr/include/stdio.h:362:12: note: expected ‘const char * restrict’ but argument is of type ‘struct _IO_FILE *’
 extern int printf (const char *__restrict __format, ...);
            ^
razer_daemon.c:117:11: warning: passing argument 1 of ‘printf’ from incompatible pointer type [-Wincompatible-pointer-types]
    printf(stderr, "Out Of Memory!\n");
           ^
In file included from razer_daemon.h:4:0,
                 from razer_daemon.c:1:
/usr/include/stdio.h:362:12: note: expected ‘const char * restrict’ but argument is of type ‘struct _IO_FILE *’
 extern int printf (const char *__restrict __format, ...);
            ^
razer_daemon.c:167:11: warning: passing argument 1 of ‘printf’ from incompatible pointer type [-Wincompatible-pointer-types]
    printf(stderr, "Out Of Memory!\n");
           ^
In file included from razer_daemon.h:4:0,
                 from razer_daemon.c:1:
/usr/include/stdio.h:362:12: note: expected ‘const char * restrict’ but argument is of type ‘struct _IO_FILE *’
 extern int printf (const char *__restrict __format, ...);
            ^
razer_daemon.c: In function ‘daemon_run’:
razer_daemon.c:240:3: warning: implicit declaration of function ‘effect3’ [-Wimplicit-function-declaration]
   effect3(daemon);
   ^
razer_daemon.c:244:3: warning: implicit declaration of function ‘razer_frame_limiter’ [-Wimplicit-function-declaration]
   razer_frame_limiter(daemon->chroma,daemon->fps);
   ^
razer_daemon.c:245:17: warning: unused variable ‘end_ticks’ [-Wunused-variable]
   unsigned long end_ticks = razer_get_ticks();
                 ^
razer_daemon.c:239:17: warning: unused variable ‘ticks’ [-Wunused-variable]
   unsigned long ticks = razer_get_ticks();
                 ^
razer_daemon.c: At top level:
razer_daemon.c:317:6: warning: conflicting types for ‘effect3’
 void effect3(struct razer_daemon *daemon)
      ^
razer_daemon.c:240:3: note: previous implicit declaration of ‘effect3’ was here
   effect3(daemon);
   ^
razer_daemon.c: In function ‘effect3’:
razer_daemon.c:320:14: warning: unused variable ‘fb’ [-Wunused-variable]
  float fr,fg,fb;
              ^
razer_daemon.c:320:11: warning: unused variable ‘fg’ [-Wunused-variable]
  float fr,fg,fb;
           ^
razer_daemon.c:320:8: warning: unused variable ‘fr’ [-Wunused-variable]
  float fr,fg,fb;
        ^
razer_daemon.c: In function ‘main’:
razer_daemon.c:995:14: warning: unused parameter ‘argc’ [-Wunused-parameter]
 int main(int argc,char *argv[])
              ^
razer_daemon.c:995:25: warning: unused parameter ‘argv’ [-Wunused-parameter]
 int main(int argc,char *argv[])
                         ^
razer_daemon.c: In function ‘dbus_handle_messages’:
razer_daemon.c:193:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
razer_daemon.c: In function ‘daemon_run’:
razer_daemon.c:248:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
In file included from razer_daemon.c:1:0:
razer_daemon.h:21:41: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_open)(struct razer_fx_render_node *render);
                                         ^
razer_daemon.h:21:41: warning: its scope is only this definition or declaration, which is probably not what you want
razer_daemon.h:22:42: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_close)(struct razer_fx_render_node *render);
                                          ^
razer_daemon.h:23:43: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_update)(struct razer_fx_render_node *render);
                                           ^
razer_daemon.h:24:46: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_key_event)(struct razer_fx_render_node *render);
                                              ^
razer_daemon.h:25:47: warning: ‘struct razer_fx_render_node’ declared inside parameter list
 typedef int (*razer_effect_dbus_event)(struct razer_fx_render_node *render);
                                               ^
razer_daemon.h:27:58: warning: ‘struct razer_daemon’ declared inside parameter list
 typedef struct razer_effect *(*razer_effect_init)(struct razer_daemon *daemon);
                                                          ^
razer_daemon.c: In function ‘dbus_handle_messages’:
razer_daemon.c:85:11: warning: passing argument 1 of ‘printf’ from incompatible pointer type [-Wincompatible-pointer-types]
    printf(stderr, "Out Of Memory!\n");
           ^
In file included from razer_daemon.h:4:0,
                 from razer_daemon.c:1:
/usr/include/stdio.h:362:12: note: expected ‘const char * restrict’ but argument is of type ‘struct _IO_FILE *’
 extern int printf (const char *__restrict __format, ...);
            ^
razer_daemon.c:117:11: warning: passing argument 1 of ‘printf’ from incompatible pointer type [-Wincompatible-pointer-types]
    printf(stderr, "Out Of Memory!\n");
           ^
In file included from razer_daemon.h:4:0,
                 from razer_daemon.c:1:
/usr/include/stdio.h:362:12: note: expected ‘const char * restrict’ but argument is of type ‘struct _IO_FILE *’
 extern int printf (const char *__restrict __format, ...);
            ^
razer_daemon.c:167:11: warning: passing argument 1 of ‘printf’ from incompatible pointer type [-Wincompatible-pointer-types]
    printf(stderr, "Out Of Memory!\n");
           ^
In file included from razer_daemon.h:4:0,
                 from razer_daemon.c:1:
/usr/include/stdio.h:362:12: note: expected ‘const char * restrict’ but argument is of type ‘struct _IO_FILE *’
 extern int printf (const char *__restrict __format, ...);
            ^
razer_daemon.c: In function ‘daemon_run’:
razer_daemon.c:240:3: warning: implicit declaration of function ‘effect3’ [-Wimplicit-function-declaration]
   effect3(daemon);
   ^
razer_daemon.c:244:3: warning: implicit declaration of function ‘razer_frame_limiter’ [-Wimplicit-function-declaration]
   razer_frame_limiter(daemon->chroma,daemon->fps);
   ^
razer_daemon.c:245:17: warning: unused variable ‘end_ticks’ [-Wunused-variable]
   unsigned long end_ticks = razer_get_ticks();
                 ^
razer_daemon.c:239:17: warning: unused variable ‘ticks’ [-Wunused-variable]
   unsigned long ticks = razer_get_ticks();
                 ^
razer_daemon.c: At top level:
razer_daemon.c:317:6: warning: conflicting types for ‘effect3’
 void effect3(struct razer_daemon *daemon)
      ^
razer_daemon.c:240:3: note: previous implicit declaration of ‘effect3’ was here
   effect3(daemon);
   ^
razer_daemon.c: In function ‘effect3’:
razer_daemon.c:320:14: warning: unused variable ‘fb’ [-Wunused-variable]
  float fr,fg,fb;
              ^
razer_daemon.c:320:11: warning: unused variable ‘fg’ [-Wunused-variable]
  float fr,fg,fb;
           ^
razer_daemon.c:320:8: warning: unused variable ‘fr’ [-Wunused-variable]
  float fr,fg,fb;
        ^
razer_daemon.c: In function ‘main’:
razer_daemon.c:995:14: warning: unused parameter ‘argc’ [-Wunused-parameter]
 int main(int argc,char *argv[])
              ^
razer_daemon.c:995:25: warning: unused parameter ‘argv’ [-Wunused-parameter]
 int main(int argc,char *argv[])
                         ^
razer_daemon.c: In function ‘dbus_handle_messages’:
razer_daemon.c:193:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
razer_daemon.c: In function ‘daemon_run’:
razer_daemon.c:248:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
::\033[32m COMPILING pez2001 fx collection [DEBUG]\033[0m
========================================================
gcc: warning: ../../lib/librazer_chroma.da: linker input file unused because linking not done
::\033[32m COMPILING pez2001 fx collection\033[0m
=====================================================
gcc: warning: ../../lib/librazer_chroma.da: linker input file unused because linking not done
=====================================================
::::::::::::::::\033[32m COMPILING examples [DEBUG]\033[0m ::::::::::::::
=====================================================
=====================================================
::::::::::::::::\033[32m COMPILING examples\033[0m ::::::::::::::
=====================================================
=====================================================
::::::::::::::::\033[32m COMPILING razer chroma kernel module\033[0m ::::::::::::::
=====================================================
In file included from include/linux/thread_info.h:11:0,
                 from ./arch/x86/include/asm/preempt.h:6,
                 from include/linux/preempt.h:18,
                 from include/linux/spinlock.h:50,
                 from include/linux/mmzone.h:7,
                 from include/linux/gfp.h:5,
                 from include/linux/slab.h:14,
                 from /tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:23:
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:654:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(mode_wave, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_wave, razer_attr_write_mode_wave);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:655:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(mode_spectrum, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_spectrum, razer_attr_write_mode_spectrum);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:656:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(mode_none, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_none, razer_attr_write_mode_none);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:657:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(mode_reactive, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_reactive, razer_attr_write_mode_reactive);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:658:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(mode_breath, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_breath, razer_attr_write_mode_breath);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:659:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(mode_custom, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_custom, razer_attr_write_mode_custom);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:660:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(mode_static, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_static, razer_attr_write_mode_static);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:661:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(temp_clear_row, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_temp_clear_row, razer_attr_write_temp_clear_row);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:662:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(set_key_row, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_set_key_row, razer_attr_write_set_key_row);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:663:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(reset, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_reset, razer_attr_write_reset);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:664:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(macro_keys, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_macro_keys, razer_attr_write_macro_keys);
        ^
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.c:665:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(set_brightness, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_set_brightness, razer_attr_write_set_brightness);
        ^
scripts/Makefile.build:264: recipe for target '/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.o' failed
make[2]: *** [/tmp/razer_blackwidow_chroma_driver/driver/razerkbd.o] Error 1
Makefile:1390: recipe for target '_module_/tmp/razer_blackwidow_chroma_driver/driver' failed
make[1]: *** [_module_/tmp/razer_blackwidow_chroma_driver/driver] Error 2
Makefile:6: recipe for target 'all' failed
make: *** [all] Error 2
```

I'm having almost the same output/error as RafTim on Fedora 22 x64. Here's the end of the output, I'll post the full output on request, I didn't because it's quite long as above....:
```
include/linux/bug.h:33:45: error: negative width in bit-field ‘<anonymous>’
 #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int:-!!(e); }))
                                             ^
include/linux/kernel.h:816:3: note: in expansion of macro ‘BUILD_BUG_ON_ZERO’
   BUILD_BUG_ON_ZERO((perms) & 2) +    \
   ^
include/linux/sysfs.h:77:12: note: in expansion of macro ‘VERIFY_OCTAL_PERMISSIONS’
    .mode = VERIFY_OCTAL_PERMISSIONS(_mode) },  \
            ^
include/linux/device.h:540:45: note: in expansion of macro ‘__ATTR’
  struct device_attribute dev_attr_##_name = __ATTR(_name, _mode, _show, _store)
                                             ^
/tmp/razer_blackwidow_chroma_driver-master/driver/razerkbd.c:665:8: note: in expansion of macro ‘DEVICE_ATTR’
 static DEVICE_ATTR(set_brightness, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_set_brightness, razer_attr_write_set_brightness);
        ^
scripts/Makefile.build:264: recipe for target '/tmp/razer_blackwidow_chroma_driver-master/driver/razerkbd.o' failed
make[2]: *** [/tmp/razer_blackwidow_chroma_driver-master/driver/razerkbd.o] Error 1
Makefile:1394: recipe for target '_module_/tmp/razer_blackwidow_chroma_driver-master/driver' failed
make[1]: *** [_module_/tmp/razer_blackwidow_chroma_driver-master/driver] Error 2
Makefile:6: recipe for target 'all' failed
```

--- a/driver/razerkbd.c
+++ b/driver/razerkbd.c
@@ -651,18 +651,18 @@ static ssize_t razer_attr_write_set_key_row(struct device *dev, struct device_at
 
 
 
-static DEVICE_ATTR(mode_wave, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_wave, razer_attr_write_mode_wave);
-static DEVICE_ATTR(mode_spectrum, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_spectrum, razer_attr_write_mode_spectrum);
-static DEVICE_ATTR(mode_none, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_none, razer_attr_write_mode_none);
-static DEVICE_ATTR(mode_reactive, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_reactive, razer_attr_write_mode_reactive);
-static DEVICE_ATTR(mode_breath, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_breath, razer_attr_write_mode_breath);
-static DEVICE_ATTR(mode_custom, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_custom, razer_attr_write_mode_custom);
-static DEVICE_ATTR(mode_static, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_mode_static, razer_attr_write_mode_static);
-static DEVICE_ATTR(temp_clear_row, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_temp_clear_row, razer_attr_write_temp_clear_row);
-static DEVICE_ATTR(set_key_row, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_set_key_row, razer_attr_write_set_key_row);
-static DEVICE_ATTR(reset, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_reset, razer_attr_write_reset);
-static DEVICE_ATTR(macro_keys, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_macro_keys, razer_attr_write_macro_keys);
-static DEVICE_ATTR(set_brightness, S_IRUGO | S_IWUSR | S_IWGRP | S_IWOTH, razer_attr_read_set_brightness, razer_attr_write_set_brightness);
+static DEVICE_ATTR(mode_wave, S_IWUSR | S_IWGRP | S_IRUGO, razer_attr_read_mode_wave, razer_attr_write_mode_wave);
+static DEVICE_ATTR(mode_spectrum, S_IWUSR | S_IWGRP | S_IRUGO, razer_attr_read_mode_spectrum, razer_attr_write_mode_spectrum);
+static DEVICE_ATTR(mode_none, S_IWUSR | S_IWGRP | S_IRUGO, razer_attr_read_mode_none, razer_attr_write_mode_none);
+static DEVICE_ATTR(mode_reactive, S_IWUSR | S_IWGRP | S_IRUGO, razer_attr_read_mode_reactive, razer_attr_write_mode_reactive);
+static DEVICE_ATTR(mode_breath, S_IWUSR | S_IWGRP | S_IRUGO, razer_attr_read_mode_breath, razer_attr_write_mode_breath);
+static DEVICE_ATTR(mode_custom, S_IWUSR | S_IWGRP | S_IRUGO, razer_attr_read_mode_custom, razer_attr_write_mode_custom);
+static DEVICE_ATTR(mode_static, S_IWUSR | S_IWGRP | S_IRUGO, razer_attr_read_mode_static, razer_attr_write_mode_static);
+static DEVICE_ATTR(temp_clear_row, S_IWUSR | S_IWGRP | S_IRUGO, razer_attr_read_temp_clear_row, razer_attr_write_temp_clear_row);
+static DEVICE_ATTR(set_key_row, S_IWUSR | S_IWGRP | S_IRUGO, razer_attr_read_set_key_row, razer_attr_write_set_key_row);
+static DEVICE_ATTR(reset, S_IWUSR | S_IWGRP | S_IRUGO, razer_attr_read_reset, razer_attr_write_reset);
+static DEVICE_ATTR(macro_keys, S_IWUSR | S_IWGRP | S_IRUGO, razer_attr_read_macro_keys, razer_attr_write_macro_keys);
+static DEVICE_ATTR(set_brightness, S_IWUSR | S_IWGRP | S_IRUGO, razer_attr_read_set_brightness, razer_attr_write_set_brightness);
 
 
 

